### PR TITLE
feat(PROXY-8,9): refactored project structure, added query params to requests

### PR DESCRIPTION
**Pull request by denStepaniuk**

### CODE REVIEW

File: app-utils/date-utils.js [hunk 1]
```diff
@@ -0,0 +1,25 @@
+const getPreviousWeekDates = () => {
+  const currentDate = new Date();
+
+  const mondayOfPreviousWeek = new Date(currentDate);
```
Comment (plutsenka) [inline app-utils/date-utils.js]: current date is always 'now' (new Date()), so it is not necessary to create variables in that way - actually, it's `new Date(new Date())`

File: app-utils/iso-utils.js [hunk 2]
```diff
@@ -0,0 +1,8 @@
+const estimatedDiameterInMeters = (min, max) => {
+  const sum = [min, max].reduce((acc, number) => acc + number);
+  return Math.floor(sum / [min, max].length);
+}
```
Comment (mdziarkach) [inline app-utils/iso-utils.js:3]:
```js
const estimatedDiameterInMeters = (min, max) => {
  return Math.floor((min + max)/ 2);
}
```

File: app-utils/date-utils.js [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  mondayOfPreviousWeek.setDate(currentDate.getDate() - currentDate.getDay() - 6);
+
+  const fridayOfPreviousWeek = new Date(currentDate);
+  fridayOfPreviousWeek.setDate(currentDate.getDate() - currentDate.getDay() - 2);
+
+  const formattedMonday = formatDate(mondayOfPreviousWeek);
+  const formattedFriday = formatDate(fridayOfPreviousWeek);
+
+  return { monday: formattedMonday, friday: formattedFriday };
```
Comment (mdziarkach) [inline app-utils/date-utils.js]: you can use vanillaJS but `date-fns` can simplify your code

File: app-utils/date-utils.js [hunk 4]
```diff
@@ -0,0 +1,25 @@
+const getPreviousWeekDates = () => {
+  const currentDate = new Date();
+
+  const mondayOfPreviousWeek = new Date(currentDate);
+  mondayOfPreviousWeek.setDate(currentDate.getDate() - currentDate.getDay() - 6);
```
Comment (plutsenka) [inline app-utils/date-utils.js]: it is a bad practice to use some abstract integer values - the better way is to store them as constants (e.g. MONDAY_OFFSET)

File: app-utils/index.js [hunk 5]
```diff
@@ -0,0 +1 @@
+console.log(`Hello world!`);
```
Comment (mdziarkach) [inline app-utils/index.js]: looks like you can remove it :)

File: app.js [hunk 6]
```diff
@@ -1,3 +1,3 @@
-const {getMeteors} = require('./proxy-app/router/router')
+const {Server} = require('./delivery/server')
 
-getMeteors();
\ No newline at end of file
+new Server().start();
```
Comment (mdziarkach) [inline app.js]:
```js
const bootstrap = () => {
  const app = new Server();
  app.start()
}

bootstrap ()
```

File: app-utils/yaml-parser.js [hunk 7]
```diff
@@ -0,0 +1,25 @@
+const fs = require('fs');
+const yaml = require('js-yaml');
+const path = require('path')
+
+const retrieveApiKey = () => {
```
Comment (plutsenka) [inline app-utils/yaml-parser.js]: Actually good approach to learn how node works with filesystem, but bad way to work with sensitive data

File: delivery/router/router.js [hunk 8]
```diff
@@ -0,0 +1,34 @@
+const express = require('express');
+const {processError} = require("./errors/error-handler");
+const {retrieveMeteorDataLastWeek, retrieveMeteorDataWithQueryParams} = require("../../use-cases/retrieve-meteors-data");
+require('dotenv').config();
+
+const router = express.Router();
+// const nasa_api_key = process.env.api_key;
```
Comment (plutsenka) [inline delivery/router/router.js]: Remove commented line, please

File: delivery/server.js [hunk 9]
```diff
@@ -0,0 +1,20 @@
+const express = require('express', 4.18);
```
Comment (plutsenka) [inline delivery/server.js]: Looks like `4.18` should be removed

File: use-cases/response-transform-utils.js [hunk 10]
```diff
@@ -0,0 +1,64 @@
+const {estimatedDiameterInMeters} = require('../app-utils/iso-utils');
+
+const transformMeteorResponse = (response) => {
+
+  const responseBody = {};
+
+  for (const key in response.data.near_earth_objects) {
+    responseBody[key] = [];
```
Comment (plutsenka) [inline use-cases/response-transform-utils.js:8]: I'd suggest to add empty line before `if` statement and remove empty line after

File: delivery/server.js [hunk 11]
```diff
@@ -0,0 +1,20 @@
...
+const {router} = require('./router/router');
+require('dotenv').config();
+
+const app = express();
+const localPort = process.env.local_port
+
+app.use(router);
+
+class Server {
```
Comment (mdziarkach) [inline delivery/server.js]: if you decide to use classes, you will need to rewrite a lot of code)

```js
class Server {
  constructor(port) {
    this.app = express();
  }

  handleRoutes() {
    app.use(router);
  }

init() {
  this.handleRoutes()
  app.listen(localPort, () => {
      console.log(`App started on port: ${localPort}`);
  })
}
}
```
and rest part of app with same approach :)

File: use-cases/response-transform-utils.js [hunk 12]
```diff
@@ -0,0 +1,64 @@
...
+                  ),
+                  is_potentially_hazardous_asteroid: meteor.is_potentially_hazardous_asteroid,
+                  close_approach_data: meteor.close_approach_data,
+                  relative_velocity: meteor.close_approach_data[0].relative_velocity.kilometers_per_second
+                }
+            )
+          }
+      );
+    }
+  }
```
Comment (plutsenka) [inline use-cases/response-transform-utils.js:28]: Just my thoughts - good idea to separate return statement with empty line - easier to read

File: use-cases/response-transform-utils.js [hunk 13]
```diff
@@ -0,0 +1,64 @@
...
+  return responseBody;
+};
+
+const retrievePotentiallyDangerousMeteors = (response) => {
+  let body =  transformMeteorResponse(response);
+  console.log(body)
```
Comment (plutsenka) [inline use-cases/response-transform-utils.js]: If you don't actually need logging (I believe you don't) - let's remove this line

File: app-utils/yaml-parser.js (hunk 7 above)
Comment (denStepaniuk) [reply app-utils/yaml-parser.js]: That was wery first implementation and attempt to get vars, now app works with .files.
I've decided to keep it as example.
Thanx for vigilence )

File: delivery/server.js (hunk 9 above)
Comment (denStepaniuk) [reply delivery/server.js]: Certainly!

File: use-cases/response-transform-utils.js (hunk 10 above)
Comment (denStepaniuk) [reply use-cases/response-transform-utils.js:8]: Done!

File: app-utils/date-utils.js (hunk 3 above)
Comment (denStepaniuk) [reply app-utils/date-utils.js]: Cool thanx ! I will remember that ) But yes it was my attempt to manage date with vanilla approach  )

File: app-utils/index.js (hunk 5 above)
Comment (denStepaniuk) [reply app-utils/index.js]: Oh no! It is requirement of very first task ) I wissh to keep it  )

**Pull request merged**